statistics/handle/lock_stats_handler.go

```diff
@@ -29,9 +27,15 @@ import (
 // - tables: table names of which will be locked.
 // Return the message of skipped tables and error.
 func (h *Handle) AddLockedTables(tids []int64, pids []int64, tables []*ast.TableName) (string, error) {
```

Comment:
After this PR, can we thoroughly detach(decouple) `StatsLock` from `StatsHandler`? Specifically, can we remove all `Handle.XXLockedXX` methods (all methods in this file `lock_stats_handler.go`) and move them into a new separate package?

Logically, if we implement these `XXLockedXX` methods on `Handler`, we think the management of locked stats is part of the responsibility of `Handler`, although `Handler` is big enough and it already has too many responsibilities.
Another way is to let the `Handler` just as a consumer of locked stats info. It doesn't update locked stats info and only reads locked stats info from the storage and reacts to it accordingly. 
I'm not sure whether this can simplify the design and implementation, what do you think of this? @hi-rustin 

![image](https://github.com/pingcap/tidb/assets/7499936/6cc742b1-aa6e-4b16-b2a7-94737adb1f01)

Reply:
I am not sure if this really helps. If we want to split the entry from the handle to stats locker. Then we need to put it in the domain or some other global structurations. I guess using the stats handle as an entry point for stats logic is fine. The problem with the handle is that we put too much business logic in the entry point. 

Right now, the lock/unlock stats handler is pretty simple:
```go
// AddLockedTables add locked tables id to store.
// - tids: table ids of which will be locked.
// - pids: partition ids of which will be locked.
// - tables: table names of which will be locked.
// Return the message of skipped tables and error.
func (h *Handle) AddLockedTables(tids []int64, pids []int64, tables []*ast.TableName) (string, error) {
	se, err := h.pool.Get()
	if err != nil {
		return "", errors.Trace(err)
	}
	defer h.pool.Put(se)

	exec := se.(sqlexec.RestrictedSQLExecutor)

	return lockstats.AddLockedTables(exec, tids, pids, tables)
}

// RemoveLockedTables remove tables from table locked array.
// - tids: table ids of which will be unlocked.
// - pids: partition ids of which will be unlocked.
// - tables: table names of which will be unlocked.
// Return the message of skipped tables and error.
func (h *Handle) RemoveLockedTables(tids []int64, pids []int64, tables []*ast.TableName) (string, error) {
	se, err := h.pool.Get()
	if err != nil {
		return "", errors.Trace(err)
	}
	defer h.pool.Put(se)

	exec := se.(sqlexec.RestrictedSQLExecutor)
	return lockstats.RemoveLockedTables(exec, tids, pids, tables)
}

// QueryTablesLockedStatuses query whether table is locked.
// Note: This function query locked tables from store, so please try to batch the query.
func (h *Handle) QueryTablesLockedStatuses(tableIDs ...int64) (map[int64]bool, error) {
	tableLocked, err := h.queryLockedTables()
	if err != nil {
		return nil, err
	}

	return lockstats.GetTablesLockedStatuses(tableLocked, tableIDs...), nil
}

// queryLockedTables query locked tables from store.
func (h *Handle) queryLockedTables() (map[int64]struct{}, error) {
	se, err := h.pool.Get()
	if err != nil {
		return nil, errors.Trace(err)
	}
	defer h.pool.Put(se)

	exec := se.(sqlexec.RestrictedSQLExecutor)
	return lockstats.QueryLockedTables(exec)
}
```

But I can try it recently to see if this helps. Do you have any suggestions on where to put the `statsLocker/statsUnlocker/statsQuerier'?

Reply:
And then in the feature, we can break the huge `StatsHandler` down to multiple smaller sub-components.
This makes our code easier to maintain (`maintaining multiple smaller and simpler components` is better than `maintaining a big and complex component`).
And these sub-components rely on the storage to synchronize information instead of relying on memory status, which is much safer.

<img width="804" alt="image" src="https://github.com/pingcap/tidb/assets/7499936/f2d30307-79ff-4040-8c94-23a5db29be3e">